crates/merod/src/main.rs

```diff
@@ -2,21 +2,36 @@ use std::env::var;
 
 use clap::Parser;
 use eyre::Result as EyreResult;
+use rand::Rng;
+use reqwest::Client;
 use tracing_subscriber::fmt::layer;
 use tracing_subscriber::prelude::*;
 use tracing_subscriber::{registry, EnvFilter};
 
 use crate::cli::RootCommand;
+use crate::version::check_for_update;
 
 mod cli;
 mod defaults;
+mod version;
 
 #[tokio::main]
 async fn main() -> EyreResult<()> {
     setup()?;
 
     let command = RootCommand::parse();
 
+    let client = Client::new();
+    let mut rng = rand::thread_rng();
+    let mut n: u8 = rng.gen();
+    if (n % 20) == 11 {
```

Comment:
also, this is a lot of boilerplate for the small thing this does, so please compress these and the client should be constructed inside `check_for_update` not here

```suggestion
    if rand::random::<u8>() % 10 == 0 {
```

same applies to meroctl